### ISSUE TITLE
Different currencies in Home and Upcoming views

### ISSUE THREAD

**Issue by sabbaghpierre** [DennisBauer/RecurringExpenseTracker#305]
**Android version(s):**

Android 14, One UI 6.1

**Device model(s):**

Samsung A14

**Last working app version (if known):**

-

**Description of problem:**

I have added expenses, and my default currency is USD. On the Home tab, the currency is correctly displayed, however on the Upcoming tab, it's showing a different currency (£).

**Screenshot or video of problem:**

![](https://imgur.com/a/expenses-bug-gA2dbis)

**Additional information:**

I believe I added one expense, and then changed the default currency. No idea if this triggered the issue.

**Comment by DennisBauer** [comment]
You can choose the currency for each expense when adding or editing it. In the overview all expenses will be shown converted to your default currency (from settings) if you use a different from the default.
On the "Upcoming" tab the expense is shown in the currency you'll pay on that day, that was actually by intention. Would you like to see the currency converted to your local currency the same as on the overview? I could think of showing the converted currency next to the original one on the upcoming tab.

**Comment by sabbaghpierre** [comment]
When adding my expenses, I manually selected USD as the price. Afterwards, I changed the default currency in the "Settings" tab to USD. In the "Home" tab, my expenses are shown correctly in USD. However, in the "Upcoming" tab they are shown in Pound Sterling, even tho I have never selected this option. I forgot what the default currency was before changing it to USD. If I edit an expense in the "Home" tab and change it to Euro for example, it will be shown correctly as Euro in the "Upcoming" tab, however, when reverting it to back to USD, it will be shown as Pound Sterling again.


![Screenshot_20240826_091442](https://github.com/user-attachments/assets/88d049f4-028d-4543-8b82-df350512e2a4)


![Screenshot_20240826_091455](https://github.com/user-attachments/assets/1466d0ba-cea2-4a49-808e-10de4d3a9cff)


![Screenshot_20240826_091457](https://github.com/user-attachments/assets/a566fd0c-71fb-42c2-b94b-febec574e408)
